pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -2042,6 +2072,28 @@ func handleNodeAgentStatusDelete(ctxArg interface{}, key string,
 	triggerPublishDevInfo(ctx)
 }
 
+func hasRealChange(x interface{}, y interface{}, excludeChgStr []string) bool {
+	diffStr := cmp.Diff(x, y)
+	diffs := strings.Split(diffStr, "\n")
+	for _, diff := range diffs {
+		var isExcluded bool
+		if strings.HasPrefix(diff, "-") || strings.HasPrefix(diff, "+") {
+			for _, exclude := range excludeChgStr {
+				if strings.Contains(diff, exclude) {
+					isExcluded = true
+					break
```

Comment:
Is there anything in the cmp package which has the ability to ignore certain fields? 
It would be nicer to not depend on the string output format of cmp.Diff.

Reply:
from my investigation, it does not appear so. they have something like if the floating point number change is smaller than a value, then let diff pass kind of things. and the structure also can be hierarchical as in our case.

Reply:
@naiming-zededa is there a better way?